plugins/datasetsplit_procedure.cc

```diff
@@ -0,0 +1,251 @@
+/** datasetsplit_procedure.cc                                                   -*- C++ -*-
+    Mathieu Marquis Bolduc, April 3rd 2017
+    This file is part of MLDB. Copyright 2017 mldb.ai inc. All rights reserved.
+*/
+
+#include "datasetsplit_procedure.h"
+#include "types/basic_value_descriptions.h"
+#include "types/distribution_description.h"
+#include "types/map_description.h"
+#include "server/dataset_context.h"
+#include "plugins/matrix.h"
+#include "server/analytics.h"
+#include "ml/value_descriptions.h"
+#include "types/any_impl.h"
+#include "jml/utils/string_functions.h"
+#include "arch/timers.h"
+#include "types/optional_description.h"
+#include "mldb/plugins/sql_config_validator.h"
+#include "mldb/plugins/sql_expression_extractors.h"
+#include "mldb/plugins/sparse_matrix_dataset.h"
+#include "mldb/server/bound_queries.h"
+
+using namespace std;
+
+
+namespace MLDB {
+
+/*****************************************************************************/
+/* DATASET SPLIT PROCEDURE CONFIG                                            */
+/*****************************************************************************/
+
+DEFINE_STRUCTURE_DESCRIPTION(SplitProcedureConfig);
+
+SplitProcedureConfigDescription::
+SplitProcedureConfigDescription()
+{
+    InputQuery labels;
+    std::vector<PolyConfigT<Dataset>> outputDatasets;
+    std::vector<float> splits;
+
+    addField("labels", &SplitProcedureConfig::labels,
+             "Row of labels to distribute among the splits while keeping their respective proportions");
+    addField("outputDatasets", &SplitProcedureConfig::outputDatasets,
+             "Configuration for output dataset. The length must match the length of splits.");
+    addField("splits", &SplitProcedureConfig::splits,
+             "Proportion of rows to put into each dataset. The sum of the input should approximate to 1.0");
+    addField("foldImportance", &SplitProcedureConfig::foldImportance,
```

Comment:
It seems to me that we should only be concerned with label distribution and not the overall split knowing that for any real-world datasets, if we split the label according to the distribution, we will get an overall split that will also be close.  This would be easier to communicate and simplify the code.